# Overview

Interactive Variant Analysis \(IVA\) is a web application for filtering, analysis and interpretation of population-scale genotype data. This interactive tool allows the identification of genes affected by deleterious variants that segregate along family pedigrees, case-control or sporadic samples.

IVA has been developed as part of the [OpenCGA](http://docs.opencb.org/display/opencga/Welcome+to+OpenCGA) project, making it easy to work with clinical and variant information stored in a Catalog and Variant storage instance.

## Main Features

* **Authenticated** and **secure** platform to query and visualise data
* Fully **customizable platform** and extensible though **pluggings**
* Multiple tools and analysis available:
  * Advanced **variant filtering**: filter your variants 
  * **Genome browser**
  * **Clinical analysis**
  * **Aggregation analysis**
* Allow to load VCF files and samples together with clinical data
* High-performance and scalable VCF and gVCF indexing
* VCF normalization and variant annotation
* Clinical interpretation analysis of samples and families

### Contact

* Ignacio Medina 

 âš  IVA is developed and maintained by the [University of Cambridge](https://www.cam.ac.uk/) and [Genomics England](https://www.genomicsengland.co.uk/). It is freely available at [**https://github.com/opencb/iv**](https://github.com/opencb/iva)

