# Overview

The Interactive Variant Analyser \(IVA\) is the web user interface for [OpenCGA](http://docs.opencb.org/display/opencga/Welcome+to+OpenCGA) that provides unprecedented features for real-time interaction with genomic data. It is suitable for any scale; from the detailed interpretation of a single genomic test through to assessing the genetic diversity of hundreds of thousands of aggregated genomes. It is not just for bioinformaticians; it provides simple and convenient access for biomedical researchers and clinical scientists as well.  

### Features

* **Variant Browser:** Efficiently browse annotated variants normalised across all samples loaded into a project. Filter variants by multiple criteria including genomic, genetic consequence, population frequency, clinical, phenotype, deleteriousness, and conservation. 
* **Case Interpreter:** Perform clinical interpretation of cases; singletons, families, cancer from panels, exomes and whole genomes, and record your interpretation findings directly into OpenCGA.  
* **Variant Analysis:** Access a variety of analysis and QC tools, run asynchronously, at sample, individual and population level. 
* **Alignment Browser:** interactively and smoothly visualise the genomic alignments \(BAM/CRAM\) associated with your samples.
* **Catalog Browser:** browse the files, samples, individuals and families associated with your studies

### Benefits

* **Authenticated** and **secure** platform to query and visualise data
* Highly **customisable;** extensible though **pluggins**

IVA has been developed as part of the [OpenCGA](http://docs.opencb.org/display/opencga/Welcome+to+OpenCGA) project, making it easy to work with clinical and variant information stored in a Catalog and Variant storage instance.

{% hint style="success" %}
  **Rich Configuration** ðŸ’¡  

Administrators can highly customise their IVA installation through a rich set of configuration options. You can easily customise the navigation bar, welcome page, filters and many other aspects of the interface.
{% endhint %}











