# Variant Browser

## Overview

The Variant browser allows you to view and filter variants for your chosen study using multiple parameters, which can be found in the left-hand menu. The filters available are: 

### Filters  <a id="IVAuserguide-Applyingfilters"></a>

### Applying filters <a id="IVAuserguide-Applyingfilters"></a>

* **Studies**
  * For studies on the within the same project \(RD38 & CG8\) you can apply AND/OR logic to filter for variants present in both studies or either one of the studies. Select the logic and tick the appropriate box fir your study of choice
* **Genomic**
  * Chromosomal location \(by genomic coordinate\)
  * Feature IDs \(genes, SNPs, transcripts\)
  * Gene disease panels \(genes included in the chosen [PanelApp ](https://panelapp.genomicsengland.co.uk/)panels\)
  * Biotype \(gene or transcript type\)
  * Variant type - SNV \(single nucleotide variant\), MNV \(multiple nucleotide variant\), CNV \(copy number variant\), SV \(structural variant\) or INDEL \(insertion/deletion\)
* **Population frequency**
  * Population frequency as calculated overall or by population group in the 1000 Genomes and gnomAD databases
* **Consequence type**
  * Consequence type SO \(Sequence ontology\) terms. You can filter for all Loss-of-Function terms here
* **Deleteriousness**
  * Deleteriousness of variant using prediction tools SIFT, Polyphen and CADD \(NB that CADD is only available in GRCh37 studies\)
* **Conservation**
  * Conservation score as calculated by PhyloP, PhastCons and Gerp.
* **Gene ontology**
  * Gene ontology terms describing gene function
* **Phenotype-disease**
  * HPO \(human phenotype ontology\) accession IDs
  * ClinVar accession IDs
  * Full-text search on HPO, ClinVar, protein domains or keywords, and some OMIM and Orphanet IDs

Once you have defined your filters, click **Search** to show the resulting variants in table format, as in the screenshot below. \(Summary format is not yet available.\)

